actions/diaspora

```diff
@@ -62,21 +63,30 @@ def set_domain_name(domain_name):
     # This did not set the domain_name
     # action_utils.dpkg_reconfigure('diaspora-common',
     #                               {'url': domain_name})
-
     # Manually changing the domain name in the conf files.
     conf_file = '/etc/diaspora.conf'
-    with open(conf_file, 'r') as conf:
-        env_vars = conf.read().splitlines()
-
-    for index, line in enumerate(env_vars):
-        if "SERVER_NAME" in line:
-            env_vars[index] = "export SERVER_NAME={}".format(domain_name)
-        if "ENVIRONMENT_URL" in line:
-            env_vars[index] = "export ENVIRONMENT_URL=http://{}:3000".format(
-                domain_name)
-
-    with open(conf_file, 'w') as conf:
-        conf.write("\n".join(env_vars))
+    aug = augeas.Augeas(flags=augeas.Augeas.NO_LOAD +
+                        augeas.Augeas.NO_MODL_AUTOLOAD)
+
+    # shell-script config file lens
+    aug.set('/augeas/load/Shellvars/lens', 'Shellvars.lns')
+    aug.set('/augeas/load/Shellvars/incl[last() + 1]', conf_file)
+    aug.load()
+
+    aug.set('/files/etc/diaspora.conf/SERVERNAME', '"{}"'.format(domain_name))
+    aug.set('/files/etc/diaspora.conf/ENVIRONMENT_URL',
+            'http://{}:3000'.format(domain_name))
+    aug.save()
+
+    # lens to edit apache2 config files
+    aug.set('/augeas/load/Httpd/lens', 'Httpd.lns')
+    aug.set('/augeas/load/Httpd/incl[last() + 1]',
+           '/etc/apache2/conf-available/*')
+    aug.load()
+
+    aug.set('/files/etc/apache2/conf-available/diaspora-plinth.conf/VirtualHost/arg', domain_name)
```

Comment:
This may cause an issue for unattended-upgrades to upgrade the plinth package. Since the file (https://github.com/freedombox/Plinth/blob/master/data/etc/apache2/conf-available/diaspora-plinth.conf) is installed by dpkg. If it is modified after install, then if it is changed in a new package version, it will require a prompt during upgrade, which will prevent unattended upgrade.

One solution is to delete https://github.com/freedombox/Plinth/blob/master/data/etc/apache2/conf-available/diaspora-plinth.conf from the plinth repository/package, and generate the file (if it doesn't already exist) during diaspora module setup.